### PR TITLE
Update ERC-7007: Move to Review

### PR DESCRIPTION
**Pull request by socathie**
We have updated our interface to support both validity (zkML) and fraud (opML) proofs. Added Rationale section. Also incorporated comments from Ethereum Magicians into Security Considerations.

### CODE REVIEW

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:4]: Your description has a lot of abbreviations and probably isn't accessible for the average cryptocurrency developer. Expanding them here might be nice, but you're also limited on characters so I'll leave this up to your judgment.

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:26]: Not a strict requirement (and more of a suggestion) but I think an illustrative use case here would be really useful. Who might be interested in using this standard, and what kind of software system would they be building?

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:32]: When talking about interfaces by name, you can use backticks (\`ERC721\`) to indicate to the linter that you're talking about code. If you want to talk about the proposal, don't use backticks (ERC-1234)

```suggestion
**Every compliant contract must implement the `IERC7007`, [`ERC721`](./eip-721.md), and [`ERC165`](./eip-165.md) interfaces.**
```

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:111]: I'd recommend using a subheader for this section:

```suggestion
### Optional Extension: Updatable
```

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:184]: This chunk here definitely belongs in Motivation, because it is justifying the proposal as a whole. The stuff in your Rationale section should explain technical choices made within the EIP itself.

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:192]: Are mint and verify function names here? If so, please enclose them in backticks (\`).

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:200]: Same comment here, but for aigcData and tokenId:

```suggestion
We specify a `mint` interface to bind the prompt and `aigcData` with `tokenId`. Notably, it acts differently in zkML and opML cases. In zkML, `mint` SHOULD make sure `verify` returns `true`. While in opML, it CAN be called before finalization. The consideration here is that, limited by the proving difficulty, zkML usually targets simple model inference tasks in practice, making it possible to provide a proof within an acceptable time frame. On the other hand, opML enables large model inference tasks, with a cost of longer confirmation time to achieve the approximate same security level. Mint until opML finalization MAY not be the best practice considering the existing optimistic protocols.
```

Comment (SamWilsn) [inline ERCS/erc-7007.md:200]: Using the ALL CAPS keyword "MAY" here creates a requirement on implementations. Use "may" instead.

File: assets/erc-7007/contracts/ERC7007Enumerable.sol
Comment (SamWilsn) [inline assets/erc-7007/contracts/ERC7007Enumerable.sol:1]: You'll need to change this to something we can legally redistribute. We prefer CC0-1.0, but I also accept MIT/Apache-2.0/Unlicense if you like.

File: assets/erc-7007/contracts/ERC7007Opml.sol
Comment (SamWilsn) [inline assets/erc-7007/contracts/ERC7007Opml.sol:1]: Change this one too.

File: assets/erc-7007/contracts/ERC7007Zkml.sol
Comment (SamWilsn) [inline assets/erc-7007/contracts/ERC7007Zkml.sol:1]: Change license here too please.

File: assets/erc-7007/contracts/IERC7007Updatable.sol
Comment (SamWilsn) [inline assets/erc-7007/contracts/IERC7007Updatable.sol:1]: License.

File: assets/erc-7007/contracts/IOpmlLib.sol
Comment (SamWilsn) [inline assets/erc-7007/contracts/IOpmlLib.sol:1]: License.

File: assets/erc-7007/contracts/MockOpmlLib.sol
Comment (SamWilsn) [inline assets/erc-7007/contracts/MockOpmlLib.sol:1]: License.

**Review by SamWilsn — CHANGES_REQUESTED** [review]

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:30]:
```suggestion
An example workflow of a zkML AIGC NFT project compliant with this proposal is as follows:
```

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:37]:
```suggestion
* AIGC-NFT smart contract - contract compliant with this proposal, with full [ERC-721](./eip-721.md) functionalities
```

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:206]: Avoid using the uppercase keywords ("SHOULD") outside the Specification section, because they create requirements on implementations. If you're explaining something, just use "should" instead.

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:218]: > The commitment should be checked inside the implementation of the `verify` function.

This is definitely a recommendation for implementations, so needs to be in the Specification section.

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:237]:
```suggestion
To address the risk of frontrunning, where an actor could potentially observe and preemptively claim a prompt during the minting process, implementers of this proposal must incorporate a secure prompt-claiming mechanism. Implementations could include time-locks, commit-reveal schemes, or other anti-frontrunning techniques to ensure equitable and secured claim processes for AIGC-NFTs.
```

File: ERCS/erc-7007.md
Comment (SamWilsn) [inline ERCS/erc-7007.md:241]:
```suggestion
In the opML scenario, it is important to consider that the `aigcData` might change during the challenge period due to disputes or updates. The updatable extension defined here provides a way to handle these updates. Implementations must ensure that updates to `aigcData` are treated as critical state changes that require adherence to the same security and validation protocols as the initial minting process. Indexers should always check for any `Update` event emission.
```

**Review by tina1998612 — APPROVED** [review]

**Review by tina1998612 — APPROVED** [review]

**Review by SamWilsn — APPROVED** [review]

**Comment by Webbdlee74** [comment]
This was air-dropped in my Coinbase wallet during onchainsummer 2023 as an ERC-20  token in the gnosis network. The status of the token or nft at the time was undeveloped. It couldn't negotiate its own value to transfer, and it couldn't be minted. I tried reaching out to base (the host of onchainsummer), and they ignored me. Coinbase, finally after 13 months, took the wallets down. I believe it was fraudulent to. It was booby trapped on other coding apps. I was told to put it on hold, I never heard anything about it.

**Pull request merged**
